src/communication/hpxreceiver.h

```diff
@@ -79,11 +79,11 @@ class HPXReceiver : public hpx::components::simple_component_base<HPXReceiver<CA
     virtual ~HPXReceiver()
     {}
 
-    void receive(std::size_t step, Cargo&& val)
+    void receive(std::size_t step, Cargo val)
     {
-        buffer.store_received(step, std::forward<Cargo>(val));
+        buffer.store_received(step, std::move(val));
     }
-    HPX_DEFINE_COMPONENT_ACTION(HPXReceiver, receive, receiveAction);
+    HPX_DEFINE_COMPONENT_DIRECT_ACTION(HPXReceiver, receive, receiveAction);
```

Comment:
Calling this action on the same node, doesn't lead to an execution of a new thread. Since this function is really tiny, you get a small performance improvement there.